pkg/cluster/cluster.go

```diff
@@ -459,6 +459,20 @@ func (c *Cluster) compareStatefulSetWith(statefulSet *appsv1.StatefulSet) *compa
 		}
 	}
 
+	if c.Statefulset.Spec.Template.Spec.PriorityClassName != statefulSet.Spec.Template.Spec.PriorityClassName {
+		match = false
+		needsReplace = true
+		needsRollUpdate = true
+		reasons = append(reasons, "new statefulset's pod priority class in spec doesn't match the current one")
+	}
+
+	if c.Statefulset.Spec.Template.Spec.Priority != statefulSet.Spec.Template.Spec.Priority {
```

Comment:
Where does this priority come from?

Reply:
From the new priority class.

the update procedure works as follows:
1. submit the new priority class
2. edit the configmap and restart the operator to  use the new class
3. wait until the operator completes the rolling upgrade

Reply:
Slightly confusing, can we limit this just to the change of the class name.

the priority we know cannot be changed and is not in scope of the operator/feature now.